posts/postRouter.js

```diff
@@ -1,27 +1,68 @@
-const express = 'express';
+const express = require('express');
 
-const router = express.Router();
+const Posts = require('./postDb.js');
 
-router.get('/', (req, res) => {
+const router = express.Router();
 
+router.get('/', async (req, res) => {
+    try {
+        const posts = await Posts.get();
+        res.status(200).json(posts);
+      } catch (error) {
+        res.status(500).json({
+          message: 'Error retrieving the posts',
+        });
+      }
 });
 
-router.get('/:id', (req, res) => {
-
+router.get('/:id', validatePostId, async (req, res) => {
+    res.status(200).json(req.post);
 });
 
-router.delete('/:id', (req, res) => {
-
+router.delete('/:id', validatePostId, async (req, res) => {
+    try {
+        const count = await Posts.remove(req.params.id);
+        if (count > 0) {
+          res.status(200).json({ message: 'The post has been deleted' });
+        } else {
+          res.status(404).json({ message: 'The post could not be found' });
+        }
+      } catch (error) {
+        res.status(500).json({
+          message: 'Error removing the post',
```

Comment:
don't forget to pass the error that is actually being caught here, it will give more details as to what might have broken.